gdpr_assist/admin/base.py

```diff
@@ -60,7 +62,11 @@ def get_urls(self):
 
     def anonymise_view(self, request):
         ids_raw = (request.POST or request.GET).get('ids')
-        objects = self.model.objects.filter(pk__in=ids_raw.split(','))
+        manager = self.model._meta.managers[0]
+        if not isinstance(manager, PrivacyManager):
+            manager = PrivacyManager._cast_class(manager)
+
+        objects = manager.filter(pk__in=ids_raw.split(','))
         verbose_name = (
             self.model._meta.verbose_name.title()
             if objects.count() == 1 else
```

Comment:
Since you will loop through objects, you can avoid a query.

```suggestion
            if len(objects) == 1 else
```